include/posets/downsets/st_backed.hh

```diff
@@ -0,0 +1,95 @@
+#pragma once
+
+#include <cassert>
+#include <iostream>
+#include <cmath>
+#include <memory>
+#include <set>
+#include <vector>
+
+#include <posets/concepts.hh>
+#include <posets/utils/sforest.hh>
+
+namespace posets::downsets {
+  // Forward definition for the operator<<s.
+  template <Vector>
+  class st_backed;
+
+  template <Vector V>
+  std::ostream& operator<<(std::ostream& os, const st_backed<V>& f);
+
+  template <Vector V> class st_backed {
+  private:
+    template <Vector V2>
+    friend std::ostream& operator<<(std::ostream& os, const st_backed<V2>& f);
+
+    int k;
+    size_t dim;
+
+    size_t root{};
+
+  public:
+    typedef V value_type;
+
+    static std::unique_ptr<utils::sforest<V>> forest;
+    static utils::sforest<V>* get_sforest() {
```

Comment:
This should not be public.  Avoid adding this method, and use `forest` instead of `get_sforest()` everywhere.